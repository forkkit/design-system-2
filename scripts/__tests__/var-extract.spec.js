/* eslint-env jest */
import { extractVarsFromCSS } from '../var-extract';
// import createHelpers from '../../../../jest.helpers';

describe('extractVarsFromCSS', () => {
  it('extracts vars from var function value with fallback', () => {
    const actual = extractVarsFromCSS(
      '.foo { background-color: var(--sds-c-icon-color-background, transparent); }'
    );
    expect(actual).toMatchSnapshot();
  });

  it('extracts vars from var function value with no fallback', () => {
    const actual = extractVarsFromCSS(
      '.foo { background-color: var(--sds-c-icon-color-background); }'
    );
    expect(actual).toMatchSnapshot();
  });
});
