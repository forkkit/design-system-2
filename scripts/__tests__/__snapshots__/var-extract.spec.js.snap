// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractVarsFromCSS extracts vars from var function value with fallback 1`] = `
Object {
  "--sds-c-icon-color-background": "transparent",
}
`;

exports[`extractVarsFromCSS extracts vars from var function value with no fallback 1`] = `
Object {
  "--sds-c-icon-color-background": "",
}
`;
